src/cdk/table/cell.ts

```diff
@@ -178,7 +178,7 @@ export class CdkHeaderCell extends BaseCdkCell {
   selector: 'cdk-footer-cell, td[cdk-footer-cell]',
   host: {
     'class': 'cdk-footer-cell',
-    'role': 'gridcell',
+    'role': 'cell',
```

Comment:
Thinking about this more, we should do the part now where the cell picks either `cell` or `gridcell` based on the table's role so that any existing apps that are assuming `grid` can switch back